File: plugin/src/test/kotlin/io/github/kaidotio/hippocampus/crypto/RailsMessageEncryptorTest.kt
package io.github.kaidotio.hippocampus.crypto

import org.junit.Assert.assertEquals
import org.junit.Test

class RailsMessageEncryptorTest {
    @Test
    fun testEncryptAES_GCM() {
        val secret = "0123456789abcdef0123456789abcdef"
        
        System.setProperty("RAILS_MASTER_KEY", secret)
        
        val message = "test"
        val encryptor = RailsMessageEncryptor()
        
        val encrypted = encryptor.encrypt(message)
        val decrypted = encryptor.decrypt(encrypted)
        
        assertEquals(message, decrypted)
    }
}

File: armyknife/internal/rails_message_encryptor/rails_message_encryptor_test.go
package rails_message_encryptor_test

import (
	"armyknife/internal/rails_message_encryptor"
	"testing"
)

func TestEncryptAES_GCM(t *testing.T) {
	secret := "0123456789abcdef0123456789abcdef"

	message := "test"
	encryptor := rails_message_encryptor.NewRailsMessageEncryptor(secret, &rails_message_encryptor.Options{})
	encrypted, err := encryptor.Encrypt(message)
	if err != nil {
		t.Errorf("Unexpected error: %v", err)
	}
	decrypted, err := encryptor.Decrypt(*encrypted)
	if err != nil {
		t.Errorf("Unexpected error: %v", err)
	}

	if *decrypted != message {
		t.Errorf("Expected %q, got %q", message, *decrypted)
	}
}
